graphql/schema.graphql

```diff
@@ -76,22 +80,21 @@ type Query {
     @checkUserAccess
     @cacheControl(scope: PRIVATE, maxAge: SHORT)
   getPaymentTerms: [PaymentTerm]
-    @cacheControl(scope: PUBLIC, maxAge: SHORT)
-    @auditAccess
+    @cacheControl(scope: PRIVATE, maxAge: SHORT)
+    @checkUserAccess
 
   getOrganizationsByEmail(email: String): [B2BOrganization]
     @checkUserAccess
     @cacheControl(scope: PRIVATE)
-    @auditAccess
 
   checkOrganizationIsActive(id: String): Boolean @cacheControl(scope: PRIVATE)
   getSalesChannels: [Channels]
-    @cacheControl(scope: PUBLIC, maxAge: SHORT)
-    @auditAccess
+    @cacheControl(scope: PRIVATE, maxAge: SHORT)
```

Comment:
What's the rationale for deciding which of these should use PRIVATE vs PUBLIC cache control? I understand that payment terms (further above in this file) seems like it should be private, depending on what it is. But sales channels and binding doesn't seem sensitive, for example, so having a public cache control seems better, since we'll be able to cache at the CDN level. Same question goes for adding @checkUserAccess for these. Aren't those available in the public VTEX rest apis today already?

I may be wrong though, just want to be sure that we don't require auth and cancel CDN caching unnecessarily.

Reply:
Good questions, but I don't know if salesChannels channels are sensitive or not. The scope `PRIVATE` was necessary because of the token, but if the information is not sensitive, we can rollback the config. For B2B, we don't have too many requests, so we need to measure if the cost of CDN is higher or not than requests for app. How does the cache control work? Is it just for the operation name or does it have another parameter in the key? I have a lot of questions to decide.

Reply:
I'm almost certain that sales channels are public, one of them is always present in the cookie by the way. So I think it's worth confirming with someone so we can decide if this needs auth or not, my guess is that it doesn't.

> For B2B, we don't have too many requests, so we need to measure if the cost of CDN is higher or not than requests for app.

The requests might be low compared to B2C, but I believe it's still worth making the right decisions here so we don't have trouble in the future. Also, from what I remember store-permissions was causing a lot of load on VBase a while ago, and requiring auth here would call store-permissions, right? Avoiding to use auth when unnecessary is good.

> How does the cache control work? Is it just for the operation name or does it have another parameter in the key?

Cache control for GraphQL APIs work per field. So if a request is made with only this `getSalesChannel` field in the root query, that value will be used as the max age for both internal caches (such as Kube Router) as well as the CDN (as long as it's PUBLIC). If a GraphQL request contains multiple query fields, the minimum max age is used, and it will be considered PRIVATE if at least one of them is configured as such. But from what I saw so far we're always sending requests with a single root query field at a time.

Reply:
The problem with VBase was another that we fixed. Even if it calls checkPermissions too many times, the problem never occurs again.

About the cache, is it the same for any account? How is the time considered for expiration?